chromadb/utils/embedding_functions.py

```diff
@@ -22,15 +23,41 @@ def __init__(self, api_key: str, model_name: str = "text-embedding-ada-002"):
             raise ValueError(
                 "The openai python package is not installed. Please install it with `pip install openai`"
             )
-        
+
         openai.api_key = api_key
         self._client = openai.Embedding
         self._model_name = model_name
+
     def __call__(self, texts: Documents) -> Embeddings:
         # replace newlines, which can negatively affect performance.
         texts = [t.replace("\n", " ") for t in texts]
         # Call the OpenAI Embedding API in parallel for each document
-        return [result["embedding"] for result in self._client.create(
-            input=texts,
-            engine=self._model_name,
-        )["data"]]
+        return [
+            result["embedding"]
+            for result in self._client.create(
+                input=texts,
+                engine=self._model_name,
+            )["data"]
+        ]
+
+
+class CohereEmbeddingFunction(EmbeddingFunction):
+    def __init__(self, api_key: str, model_name: str = "large"):
+        try:
+            import cohere
+        except ImportError:
+            raise ValueError(
+                "The cohere python package is not installed. Please install it with `pip install cohere`"
+            )
+
+        self._client = cohere.Client(api_key)
+        self._model_name = model_name
+
+    def __call__(self, texts: Documents) -> Embeddings:
+        # replace newlines, which can negatively affect performance.
+        texts = [t.replace("\n", " ") for t in texts]
```

Comment:
Is this also the case for cohere embeddings? Interesting, is there discussion of this phenomenon somewhere?

Reply:
No, as far as I know this is officially not mentioned anywhere or at least I have not come across but I assumed this to be a general case for cohere too. I'll confirm with someone at Cohere to help with this.

Reply:
Asked on cohere discord directly quoting the reply from one of their staff members - "_we haven't done a thorough analysis either, it's probably worth trying yourself to see, but I suspect they should be okay! there's nothing intrinsic about our method that would make it perform worse on new lines _ "

Reply:
May I know your thoughts on this? @atroyn

Reply:
Let's take out whitespace replacement until we learn more, this is a good experiment to run!

Reply:
Sure sounds right. Will need to see how it affects the performance.